core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/S3BatchSource.java

```diff
@@ -72,25 +84,47 @@ private static String updateFileSystemProperties(@Nullable String fileSystemProp
   public static class S3BatchConfig extends FileBatchConfig {
     @Description(ACCESS_ID_DESCRIPTION)
     @Macro
+    @Nullable
     private final String accessID;
 
     @Description(ACCESS_KEY_DESCRIPTION)
     @Macro
+    @Nullable
     private final String accessKey;
 
-    public S3BatchConfig(String referenceName, String accessID, String accessKey, String path) {
-      this(referenceName, accessID, accessKey, path, null, null, null, null, null, false, false);
+    @Description(AUTHENTICATION_METHOD)
```

Comment:
IAM role is not needed when accessing S3 bucket using IAM role based authentication from hadoop.We just need to omit the key properties.
https://hadoop.apache.org/docs/r3.0.0-alpha1/hadoop-aws/tools/hadoop-aws/index.html

IAMRole configuration is needed in redshift copy command while accessing S3.
http://docs.aws.amazon.com/redshift/latest/dg/t_loading-tables-from-s3.html